lib/rules/no-unused-vars.js

```diff
@@ -433,7 +433,7 @@ module.exports = {
                     ) ||
                 (
                     parent.type === "UpdateExpression" &&
-                    grandparent.type === "ExpressionStatement"
+                   (grandparent.type === "ExpressionStatement" || grandparent.type === "SequenceExpression")
```

Comment:
This would be a false positive:

```js
/* eslint no-unused-vars: error */

let x = 1, y;

do {
  y = (0, x++); // false positive on `x`
} while (foo(y));

```

I think we should more precisely define what should be fixed.

Reply:
Ah yes, any suggestions?

Reply:
@mdjermanovic looking for your feedback here

Reply:
after some investigation, I found it was not related to comma. https://github.com/eslint/eslint/issues/14325#issuecomment-831327960

let's discuss in that thread. :)

Reply:
The `grandparent.type === "ExpressionStatement"` condition aims to check if the value of `parent` is unused. It covers only expressions in statement positions.

If we want to expand this check to positions in sequence expressions, we could make `function isUnusedExpression(node)`:

* if `node.parent` is `"ExpressionStatement"`, then `true`
* if `node.parent` is `"SequenceExpression"` and `node` is not the last expression in the sequence, then `true`.
* if `node.parent` is `"SequenceExpression"` and `node` is the last expression, then go up recursively.
* otherwise `false`

and then replace `grandparent.type === "ExpressionStatement"` with `isUnusedExpression(parent)`.

That would catch `x` as an unused variable in the following example:

```js
var x = 0;

x++, 0;
0, x++;
0, (1, x++);
foo = (x++, 0); // foo = (0, x++); would be treated as used
foo = ((0, x++), 0);
```

Reply:
I pushed changes implementing `function isUnusedExpression(node)`, can you check if it's going in the right direction?

Reply:
Yes, that's the direction I was thinking of, thanks!

Reply:
@aladdin-add 

> after some investigation, I found it was not related to comma. [#14325 (comment)](https://github.com/eslint/eslint/issues/14325#issuecomment-831327960)
> 
> let's discuss in that thread. :)

I think it makes sense to account for sequence expressions in `no-unused-vars` (i.e. I think this PR makes sense), I'll leave a comment on https://github.com/eslint/eslint/issues/14325#issuecomment-831327960.